modules/op-configuring-ssh-authentication-for-git.adoc

```diff
@@ -0,0 +1,97 @@
+// This module is included in the following assembly:
+//
+// *openshift-docs/cicd/pipelines/authenticating-pipelines-using-git-secret.adoc
+
+[id="op-configuring-ssh-authentication-for-git_{context}"]
+= Configuring SSH authentication for Git
+
+For SSH authentication, {pipelines-shortname} creates a `~/.ssh/config` file containing the SSH private key specified in the secret, before executing any steps in a task run or a pipeline run. When the steps execute, {pipelines-shortname} uses the key to retrieve the pipeline resources specified in a run.
+
+[NOTE]
+====
+It is recommended to use SSH based authentication over basic authentication.
+====
+
+[discrete]
+.Procedure
+
+To configure a Git secret of the `ssh-auth` type:
+
+. Generate a link:https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent[SSH private key], or copy an existing private key which is usually available in the `~/.ssh/id_rsa` file.
+. In the `secret.yaml` file, specify the username and password required to access the target Git repository.
```

Comment:
Do we need to update this sentence? I don't see a username or password element in the following example.

Reply:
Thanks for catching this mistake. I copied the format from the basic auth file, and then forgot to edit his sentence. You saved me from an embarrassment.

> Also consider mentioning example values such as ~/.ssh/known_hosts
The problem with this example is that `~/.ssh/known_hosts` points to a file, whereas the `ssh-privateKey` expects a string (or an actual private key), instead of a file name containing the private key.